pyteal/ast/abi/array_static.py

```diff
@@ -78,6 +78,21 @@ def __init__(self, array_type_spec: StaticArrayTypeSpec[T, N]) -> None:
     def type_spec(self) -> StaticArrayTypeSpec[T, N]:
         return cast(StaticArrayTypeSpec[T, N], super().type_spec())
 
+    def decode(
+        self,
+        encoded: Expr,
+        *,
+        startIndex: Expr = None,
+        endIndex: Expr = None,
+        length: Expr = None,
+    ) -> Expr:
+        if endIndex is not None or length is not None:
+            raise TealInputError(
+                "Expected None for endIndex and length on static array"
+            )
+
+        return super().decode(encoded, startIndex=startIndex, length=self.length())
```

Comment:
```suggestion
        return super().decode(encoded, startIndex=startIndex, length=self.byte_length())
```

Maybe this fixes the bug I observed?

Reply:
sorry `byte_length_static`

Reply:
...of the associated typespec